daemon/mount.c

```diff
@@ -326,13 +327,29 @@ mount_private_tmp(void)
 		ERROR_ERRNO("Could not unshare host mount ns!");
 		return -1;
 	}
-	if (umount("/tmp") < 0 && errno != ENOENT) {
-		ERROR_ERRNO("Could not umount /tmp");
+	if (file_is_mountpoint("/tmp")) {
+		if (umount("/tmp") < 0 && errno != ENOENT) {
+			ERROR_ERRNO("Could not umount /tmp");
+			return -1;
+		}
+	}
+	if (mount(NULL, "/", NULL, MS_REC | MS_PRIVATE, NULL) < 0) {
```

Comment:
I think in hosted mode it would be easier for the user to set the propagation type to shared

Reply:
the intention of this is that the host mount namespace is kept clean of any
container mounts.